packages/cli/src/actions/PullAction.ts

```diff
@@ -0,0 +1,112 @@
+import { ExperimentApiService } from "../services/experiment/ExperimentApiService";
+import { ExperimentConfigModel } from "../config/ExperimentsConfig";
+import { ExperimentFlagComparator } from "../services/experiment/ExperimentFlagComparator";
+import { ComparisonResultSymbol, ICON_RETURN_ARROW, ICON_SUCCESS, ICON_WARNING_W_TEXT } from "../ui/icons";
+import { ComparisonResult } from "../comparison/ComparisonResult";
+import { cloneDeep } from "lodash";
+import { loadLocalConfiguration, saveYamlToFile } from "../config/AmplitudeConfigYamlParser";
+import { BaseAction } from "./BaseAction";
+
+export interface PullActionOptions {
+  config: string;
+  experimentManagementApiKey?: string;
+  experimentDeploymentId?: string;
+}
+
+export class PullAction extends BaseAction {
```

Comment:
this pull is different from the pull action we have now I guess? Just the name is same?

Reply:
Yep, in the Alpha SDK splits the "pull" command into two parts. `cli pull` only downloads the configuration, schema. `cli build` will generated the SDK source code.

Reply:
OK. Would you consider combining them later?

Reply:
This is a feature of the new SDK/Codegen. By breaking up 1) downloading the tracking plan and 2) generating the source code, customers will be able to play around with codegen without needing to go to the Data/Experiment website. This also makes it possible to host the codegen in a public sandbox without the need for a dummy tracking plan.